pkg/channel/fanout/fanout_message_handler.go

```diff
@@ -164,5 +164,8 @@ func (f *MessageHandler) makeFanoutRequest(ctx context.Context, message binding.
 	if sub.DeadLetterSinkURI != nil {
```

Comment:
Does this part makes sense? I'm not sure I understood how to use the duck type `Destination` and I don't get if the `Subscribable` should contain the `DeadLetterSinkURI` in `sub.Delivery.DeadLetterSink`,  `sub.DeadLetterSinkURI` or both

Reply:
https://github.com/knative/eventing/pull/2876 should address it